.github/actions/setup-cross-build/action.yml

```
@@ -0,0 +1,67 @@
+name: Setup Cross Build Environment
+description: Prepare cross-compile environment (Docker image, macOS deps, Rust toolchain)
+
+inputs:
+  target:
+    description: Rust target triple to build for
+    required: true
+  cross_image:
+    description: Local Docker image tag to build and load (optional)
+    required: false
+    default: ""
+  os:
+    description: Runner OS (unused, kept for compatibility)
+    required: false
+    default: ""
+  rust_version:
+    description: Rust toolchain to install
+    required: false
+    default: "stable"
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout fuel-core for Docker files
+      if: ${{ inputs.cross_image != '' }}
+      uses: actions/checkout@v3
```

i think we can checkout once:
```suggestion
    - name: Checkout fuel-core sources
      if: ${{ inputs.cross_image != '' || runner.os == 'macOS' }}
      uses: actions/checkout@v4
```

---

good catch!